app/views/govuk_publishing_components/components/_date_input.html.erb

```diff
@@ -0,0 +1,85 @@
+<%
+  id ||= "input-#{SecureRandom.hex(4)}"
+  name ||= nil
+  items ||= nil
+
+  legend_text ||= nil
+  hint ||= nil
+  error_message ||= nil
+  describedby ||= nil
+
+  has_error = error_message
+
+  css_classes = %w(gem-c-date-input govuk-date-input)
+  form_group_css_classes = %w(govuk-form-group)
+  form_group_css_classes << "govuk-form-group--error" if has_error
+
+  hint_id = "hint-#{SecureRandom.hex(4)}"
+  error_id = "error-#{SecureRandom.hex(4)}"
+
+  aria_described_by ||= nil
+  if hint || has_error || describedby
+    aria_described_by = []
+    aria_described_by << hint_id if hint
+    aria_described_by << error_id if has_error
+    aria_described_by << describedby if describedby
+    aria_described_by = aria_described_by.join(" ")
+  end
+
+%>
+
+<% date_input_items = items ? items : date_input_items = [
+  { :name => "day", :width => 2 },
+  { :name => "month", :width => 2 },
+  { :name => "year", :width => 4 }
+] %>
+
+<%= content_tag :div, class: form_group_css_classes do %>
+  <% fieldset_content = capture do %>
+    <% if hint %>
+      <%= render "govuk_publishing_components/components/hint", {
+        id: hint_id,
+        text: hint,
+        margin_bottom: 2
+      } %>
+    <% end %>
+
+    <% if error_message %>
+      <%= render "govuk_publishing_components/components/error_message", {
+        id:  error_id,
+        text: error_message
+      } %>
+    <% end %>
+
+    <%= tag.div class: css_classes, id: id do %>
+      <% date_input_items.each do |item| %>
+        <%= tag.div class: "govuk-date-input__item" do %>
+          <%= render "govuk_publishing_components/components/input", {
+            label: {
+              text: item[:label] || item[:name].capitalize
+            },
+            error_style: has_error,
+            name: name ? (name + "[" + item[:name] + "]") : item[:name],
```

Comment:
Could we do this automatically for the defaults? Since overriding the items requires you to specify a name anyway, I think we can express this helpful behaviour less magically :-).